VoteCounter/src/GUI.cs

```diff
@@ -27,48 +27,54 @@ private void btnGenerateVoteCount_Click(object sender, EventArgs e)
 
             if (Int32.TryParse(txtStartingPost.Text, out startingPostNumber) && Int32.TryParse(txtEndingPost.Text, out endingPostNumber))
             {
-                PostList postList = new PostList(url, startingPostNumber, endingPostNumber);
-
                 //make a List
                 List<Player> players = new List<Player>();
 
                 //set the players
                 foreach (DataGridViewRow row in listPlayers.Rows)
                 {
-                    List<string> nicks = new List<string>();
-                    string name = "";
-                    for (int i = 0; i < listPlayers.Columns.Count; i++)
+                    Player newPlayer = new Player("");
+
+                    foreach (DataGridViewColumn column in listPlayers.Columns)
                     {
-                        //if under column Names
-                        if (listPlayers.Columns[i].Name.Equals("Name")
-                            && row.Cells[i].Value != null)
-                            name = row.Cells[i].Value.ToString();
-
-                        //if under column Nicknames
-                        if (listPlayers.Columns[i].Name.Equals("Nicknames")
-                            && row.Cells[i].Value != null)
+                        if (row.Cells[column.Index].Value != null)
                         {
-                            if (row.Cells[i].Value.ToString().Length > 0)
+                            //if under column Names
+                            if (column.Name.Equals("Name"))
                             {
-                                //parse the string, seperate each name, then put it in the List
-                                string[] words = row.Cells[i].Value.ToString().Split(',');
-                                //then add them
-                                foreach (string s in words)
+                                //if name is valid
+                                if (row.Cells[column.Index].Value.ToString() != ""
+                                    && row.Cells[column.Index].Value != null)
                                 {
-                                    nicks.Add(s);
+                                    newPlayer.mainName = row.Cells[column.Index].Value.ToString();
                                 }
-                                //its adding a blank to the end, so I'll just remove that manually
-                                if (nicks.Contains(""))
+                            }
+                            //if under column Nicknames
+                            else if (column.Name.Equals("Nicknames"))
+                            {
+                                if (row.Cells[column.Index].Value.ToString().Length > 0
+                                    && row.Cells[column.Index].Value != null)
```

Comment:
row.Cells[column.Index].Value can't be null here since it's already been checked as != null above.